doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -716,24 +710,13 @@ The following example script illustrates all these features:
    You tried to apply a tactic but no goals were under focus.
    Using :n:`@bullet` is  mandatory here.
 
-.. FIXME: the :noindex: below works around a Sphinx issue.
-   (https://github.com/sphinx-doc/sphinx/issues/4979)
-   It should be removed once that issue is fixed.
-
 .. exn:: No such goal. Try unfocusing with %}.
-   :noindex:
 
    You just finished a goal focused by ``{``, you must unfocus it with ``}``.
 
-Mandatory Bullets
-~~~~~~~~~~~~~~~~~
-
-Using :opt:`Default Goal Selector` with the ``!`` selector forces
-tactic scripts to keep focus to exactly one goal (e.g. using bullets)
-or use explicit goal selectors.
-
-Set Bullet Behavior
-~~~~~~~~~~~~~~~~~~~
+.. note:: Use :opt:`Default Goal Selector` with the ``!`` selector forces
+   tactic scripts to keep exactly one goal focused (e.g. using bullets)
+   or use explicit goal selectors.
```

Comment:
```suggestion
   or to use explicit goal selectors.
```

Reply:
No, should not be the infinitive.  Here's what I put (with "to force"):

```
.. note:: Use :opt:`Default Goal Selector` with the ``!`` selector to force
   tactic scripts to keep exactly one goal focused (e.g. using bullets)
   or use explicit goal selectors.
```

Reply:
But is it clear then that the parentheses are this way:

>Use :opt:`Default Goal Selector` with the ``!`` selector to [force tactic scripts to keep exactly one goal focused (e.g. using bullets) or use explicit goal selectors].

and not this way:

>[Use :opt:`Default Goal Selector` with the ``!`` selector to force tactic scripts to keep exactly one goal focused (e.g. using bullets)] or [use explicit goal selectors].

I find it ambiguous when reading this.

Reply:
It's not very clear.  How about something more like:

Use `Set Default Goal Selector !` to ensure that each tactic is applied to a single goal at a time.  This will cause a failure if more than one goal is in focus.  Less conveniently, you can also add goal selectors such as `1 :` where needed.

OTOH, the default goal selector is `1` (the initial setting of Default Goal Selector), which also ensures that the tactic is only applied to a single goal at a time (unless you have a selector). So maybe this is just a useless note that should be dropped?

Reply:
No, your new wording is less good I think. Maybe you have misunderstood the point of `!` (at least this is what I gather from your question).

The point of `Set Default Goal Selector !` is not to ensure that each tactic is applied to a single goal, not at all: it doesn't even prevent applying tactics to several goals (with goal selectors).

`Set Default Goal Selector !` forces the use of focusing mechanisms (bullets, braces) and goal selectors so that it is always explicit to which goal(s) a tactic is applied.

Does this help?